.github/workflows/sync.yml

```
@@ -0,0 +1,194 @@
+name: Sync
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+
+env:
+  BOT_USERNAME: "exercism-bot"
+  GIT_USERNAME: "Exercism Bot"
+  GIT_EMAIL: "66069679+exercism-bot@users.noreply.github.com"
+
+jobs:
+  fetch-tracks:
+    name: Fetch & fork tracks
+    runs-on: ubuntu-20.04
+
+    outputs:
+      tracks: ${{ steps.fetch-tracks.outputs.result }}
+
+    steps:
+      - name: Fetch list of tracks
+        id: fetch-tracks
+        uses: actions/github-script@47f7cf65b5ced0830a325f705cad64f2f58dddf7
+        with:
+          script: |
+            const trackRepos = await github.paginate(github.search.repos, {
+              q: 'org:exercism+topic:exercism-track'
+            })
+
+            //return trackRepos.flatMap(({ full_name }) => [full_name])
+
+            // TODO: Before merge, remove this and uncomment the line above
+            return ['exercism/julia']
+
+      - name: Determine tracks needing to be forked
+        id: required-forks
+        uses: actions/github-script@47f7cf65b5ced0830a325f705cad64f2f58dddf7
+        with:
+          github-token: ${{ secrets.BOT_PERSONAL_ACCESS_TOKEN }}
+          script: |
+            const botRepos = await github.paginate(github.repos.listForAuthenticatedUser, {
+              affiliation: 'owner'
+            })
+
+            // Replacing the bot username with exercism is slightly hacky, but necessary
+            // for the comparison below. Unfortunately, GitHub's API response does not seem
+            // to contain any reference to the upstream repo of the fork.
+            const botForks = botRepos.flatMap(({ full_name, fork }) => fork ? [full_name.replace(process.env.BOT_USERNAME, 'exercism')] : [])
+
+            const tracks = ${{ steps.fetch-tracks.outputs.result }}
+
+            return tracks.filter(track => !botForks.includes(track))
+
+      - name: Fork repos that haven't been forked yet
+        if: steps.required-forks.outputs.result != '[]'
+        uses: actions/github-script@47f7cf65b5ced0830a325f705cad64f2f58dddf7
+        with:
+          github-token: ${{ secrets.BOT_PERSONAL_ACCESS_TOKEN }}
+          script: |
+            const trackReposFullNames = ${{ steps.required-forks.outputs.result }}
+            const trackRepos = trackReposFullNames.map((fullName) => fullName.split('/')[1])
+
+            for (repo of trackRepos) {
```

Is this missing something?

```suggestion
            for (const repo of trackRepos) {
```